manager/src/main/java/org/openremote/manager/dashboard/DashboardResourceImpl.java

```diff
@@ -30,31 +31,66 @@ public DashboardResourceImpl(TimerService timerService, ManagerIdentityService i
 
     @Override
     public Dashboard[] getAllRealmDashboards(RequestParams requestParams, String realm) {
-
+        boolean publicOnly = true;
+        if(realm == null) {
+            throw new WebApplicationException(BAD_REQUEST);
+        }
         try {
+            // Realm should be enabled. Also takes unauthenticated users into count.
             if(!isRealmActiveAndAccessible(realm)) {
                 throw new WebApplicationException(FORBIDDEN);
             }
-            return this.dashboardStorageService.findAllOfRealm(realm, getUserId());
+            // if not authenticated, or having no INSIGHTS access, only fetch public dashboards
+            if(isAuthenticated()) {
+                publicOnly = (!hasResourceRole(ClientRole.READ_INSIGHTS.getValue(), Constants.KEYCLOAK_CLIENT_ID));
+            }
+            return this.dashboardStorageService.query(null, realm, getUserId(), publicOnly, false);
 
         } catch (IllegalStateException ex) {
-            throw new WebApplicationException(ex, BAD_REQUEST);
+            ex.printStackTrace();
+            throw new WebApplicationException(ex, INTERNAL_SERVER_ERROR);
         }
     }
 
     @Override
     public Dashboard get(RequestParams requestParams, String dashboardId) {
-        return this.dashboardStorageService.get(dashboardId);
+        boolean publicOnly = true;
+        String realm = getRequestRealmName();
+        try {
```

Comment:
As you have a null check on realm in other methods with `BAD_REQUEST` if null it should be here also for consistency